include/base/libmesh_common.h

```diff
@@ -586,6 +586,17 @@ inline Tnew libmesh_cast_int (Told oldvar)
 #define libmesh_fallthrough() ((void) 0)
 #endif
 
+// Recursive tuple scheme
+template <size_t I, typename T>
+struct libmesh_tuple_n {
+    template< typename...Args> using type = typename libmesh_tuple_n<I-1, T>::template type<T, Args...>;
+};
```

Comment:
This doesn't match our usual indentation style for classes/structs (opening curly brace on separate line + indent two spaces).

Reply:
I don't think libmesh_common.h is the right place for this... I'd say either make a new header file or maybe put it in `type_vector.h` since `type_tensor.h` and `type_n_tensor.h` both include that header. Also the new structs should be in  the libMesh namespace IMO.

Reply:
They are in the libMesh namespace, but yes, moving to another location makes sense to me

Reply:
> They are in the libMesh namespace,

Ah, yeah, I thought we just had a bunch of macros in that file, but the whole thing is wrapped in a namespace. In that case, the name of the struct doesn't really need to have `libmesh_` prepended.